fastly/paginator.go

```diff
@@ -29,3 +37,178 @@ type PaginatorKVStoreEntries interface {
 	Keys() []string
 	Err() error
 }
+
+func NewPaginator[T ACLEntry | DictionaryItem | Service](client *Client, input *ListInput, path string) Paginator[T] {
+	return &ListPaginator[T]{
+		client: client,
+		input:  input,
+		path:   path,
+	}
+}
+
+type ListInput struct {
+	// Direction is the direction in which to sort results.
+	Direction string
+	// Page is the current page.
+	Page int
+	// PerPage is the number of records per page.
+	PerPage int
+	// Sort is the field on which to sort.
+	Sort string
+}
+
+// Paginator represents a generic paginator.
+type Paginator[T ACLEntry | DictionaryItem | Service] interface {
+	HasNext() bool
+	Remaining() int
+	GetNext() ([]*T, error)
+}
+
+// ListPaginator implements the generic Paginator[N] interface.
+type ListPaginator[T ACLEntry | DictionaryItem | Service] struct {
```

Comment:
I would probably have this take `any` rather than specific types.  What's the benefit to constraining it?

(Also, I think it's more idiomatic to constrain to an interface that contains the type sets, like `ordered` below.)

Reply:
Yup. Was originally `any` but code went (and still is) going through iterations. Pushed up as a draft as I'm having trouble figuring out how best to implement a generic sort as original code would use a stable sort that was unique fields per type. If you have thoughts on that I'd love to hear them 🙂